fpsim/sim.py

```diff
@@ -120,6 +120,7 @@ def __init__(self, pars=None, location=None, label=None, track_children=False, r
         self.track_children = track_children
         self.regional = regional
         self.ti = None  # The current timestep of the simulation
+        self.pars['tperyear'] = self.tperyear
```

Comment:
Where's `tperyear` defined for the first time? Is it user defined? 

I would suggest that this parameter is defined in `parameters.py`, under basic parameters or perhaps settings and set to `None`
![image](https://github.com/user-attachments/assets/8fa6cbb0-8703-490a-8f96-a42de704fe4f)

The actual value of `tperyear` can be defined in `kenya.py` and would be updated as the other location-specific parameters. With that change, I believe we would not have to pass `self.tperyear` to the `self.people.update()` method as an argument. Internally, the class `People` could assess whether `tperyear == None` and decide whether to create the longitudinal attributes accordingly. Hope that makes sense.

Reply:
Thank you for the feedback, Paula! tperyear is defined initially in base.py (as a property function, similarly to 'npts'). It's used to determine how many columns to create for the circular buffer for longitudinal params. I originally was going to put it in parameters.py but it requires the npts param to do the calculation, which from what I recall hadn't yet been defined yet when the default pars are set. Technically the tperyear are the same (=13) regardless of the start and end years, so I could either hardcode it (probably not the best way in case the logic for this ever changes) or actually just calculate it in parameters.py using the same logic as in base.py to calculate the npts and then the tperyear. Thoughts?

Reply:
> Technically the tperyear are the same (=13) regardless of the start and end years,
This is assuming a timestep of 1 month, right? 

And the circular buffer needs to always be of length 1 year and 1 month, expressed in number of timesteps? Or does it have to be of length 1 year + 1 timestep?
  
If we had a timestep of 0.5 months (I know would be unusual, but since timestep can be any float ...) would the circular buffer size be 26 or 25?

Reply:
>  so I could either hardcode it (probably not the best way in case the logic for this ever changes) or actually just calculate it in parameters.py using the same logic as in base.py

You're right, we do not want to hardcode this, and for the time being I'd leave as you have implemented it. 

It's usually trick to decide where to put these calculations. `npts`, and `tperyear` are "derived" parameters, and could easily exist in parameters.py and add them to `pars` dictionary _after_ some sort of validation to make sure everything is consistent before we even initialise the sim. On the other hand, they are time parameters relevant to the simulation itself. Probably something to consider for future refactoring.

Reply:
> It's usually tricky to decide where to put these calculations. `npts`, and `tperyear` are "derived" parameters, and could easily exist in parameters.py and add them to `pars` dictionary _after_ some sort of validation to make sure everything is consistent before we even initialise the sim. On the other hand, they are time parameters relevant to the simulation itself. Probably something to consider for future refactoring.
Agreed. I'll leave it in base.py for now since it's a time parameter (even though it ends up being the same value despite start/end years changing). I can add a TODO next to the definition as a reminder in case we decide to refactor the placement of the derived parameters.

Reply:
> > Technically the tperyear are the same (=13) regardless of the start and end years,
> 
> ☝️ This is assuming a timestep of 1 month, right?
> 
> And the circular buffer needs to always be of length 1 year and 1 month, expressed in number of timesteps? Or does it have to be of length 1 year + 1 timestep?
> 
> If we had a timestep of 0.5 months (I know would be unusual, but since timestep can be any float ...) would the circular buffer size be 26 or 25?

Yes, based on months per year (see npts property in base.py)
![image](https://github.com/user-attachments/assets/e94047b6-6ca4-4925-863a-516a139a4d08)

I have the circular buffer set up to be the number of timesteps (not necessarily months) per year, rounded up to the nearest integer. 

So if we had a timestep of .5 months, the circular buffer size would be 25 as the actual tiperyear would be 24.x but we round up to 25 to accommodate all the timesteps per year in the circular buffer.

Let me know if you have any more questions/comments, or if this all sounds ok, feel free to resolve the conflicts (thank you!) and merge the branch.

Reply:
This all sounds good @emilydriano! I'll go ahead work on merging this branch into `rc2.0-methodtime` :crossed_fingers: